bitmapcontainer.go

```diff
@@ -109,13 +109,56 @@ func (bcsi *bitmapContainerShortIterator) next() uint16 {
 func (bcsi *bitmapContainerShortIterator) hasNext() bool {
 	return bcsi.i >= 0
 }
+
 func newBitmapContainerShortIterator(a *bitmapContainer) *bitmapContainerShortIterator {
 	return &bitmapContainerShortIterator{a, a.NextSetBit(0)}
 }
+
 func (bc *bitmapContainer) getShortIterator() shortIterable {
 	return newBitmapContainerShortIterator(bc)
 }
 
+type bitmapContainerManyIterator struct {
+	ptr    *bitmapContainer
+	base   int
+	bitset uint64
+}
+
+func (bcmi *bitmapContainerManyIterator) nextMany(hs uint32, buf []uint32) int {
+	n := 0
+	base := bcmi.base
```

Comment:
go didn't seem to be able to optimise these away in the inner loop, but perhaps i was missing something (similar idiom in the other functions)

Reply:
The likely reason is that your referring to an pointer (*bcmi). The value that your referring might change anytime as it’s visible to other threads / goriutines.